File: Zadania Szkopul (C++ i Java)/Zbieranie informacji/Zbieranie_informacji.cpp
#include <bits/stdc++.h>

using namespace std;
int kom,inf,p=1,wynik; // kom - liczpa komputeruf w żendzie oraz fragmentuf informacyji; inf - posiadane informacje; p - pozycja
bool kier=1; // kier - kierunek ruchu, 1 - prawo, 0 - lewo;
int Dro[1005]; // tablica z wartościami potrzebnymi do dostania się do kolejnego laptopa;
int main()
{
	Dro[0]=-20;
	scanf("%d",&kom);
	for(int i=1;i<=kom;i++)
	{
		scanf("%d",&Dro[i]);
	}
	while(inf<kom)
	{
		if(Dro[p]<=inf and Dro[p]>=0)
			{
				Dro[p]=-10;
				inf++;
				if(kier==1)
					p++;
				else
					p--;
					
				printf("P:%d,Inf %d, 1, wynik %d\n",p,inf,wynik);
			}
			else
			{
				if(kier==1)
					p++;
				else
					p--;
				printf("P:%d,Inf %d, 2, wynik %d\n",p,inf,wynik);
			}
		if(inf<kom)
			if(p==kom or p==0)
			{
				kier=(1-kier);
				wynik++;
				printf("P:%d,Inf %d, 3, wynik %d\n",p,inf,wynik);
			}
		
	}
	printf("%d",wynik);
}

File: Zadania Szkopul (C++ i Java)/Zbieranie informacji/Zbieranie_informacji.java
import java.util.Scanner;

public class Zbieranie_informacji {
    public static void main(String[] args) {
        int kom,inf=0,p=1,wynik=0; // kom - liczpa komputeruf w żendzie oraz fragmentuf informacyji; inf - posiadane informacje; p - pozycja
        int kier=1; // kier - kierunek ruchu, 1 - prawo, 0 - lewo;
        int Dro[] = new int[1005]; // tablica z wartościami potrzebnymi do dostania się do kolejnego laptopa;
        Scanner czytaj = new Scanner(System.in);
        Dro[0] = -20;
        kom = czytaj.nextInt();
        for(int i=1; i<=kom; i++){
            Dro[i] = czytaj.nextInt();
        }
        czytaj.close();
        while(inf < kom){
            if(Dro[p] <= inf && Dro[p] >=0){
                Dro[p] = -10;
                inf++;
                if(kier == 1){
                    p++;
                }
                else{
                    p--;
                }
            }
            else{
                if(kier == 1){
                    p++;
                }
                else{
                    p--;
                }
            }
            if(inf < kom){
                if(p == kom || p == 0){
                    kier = (1 - kier);
                    wynik++;
                }
            }
        }
        System.out.println(wynik);
        
    }
}
